### ISSUE TITLE
Menu segment title doesn't render as a header 

### ISSUE THREAD

**Issue by niamhmadden24** [Sage/carbon#6010]
### Current behaviour

The segment titles within menus are not rendered as headings. This is an accessibility issue and fails against 4.1.2 of [Web Content Accessibility Guidelines (WCAG) 2.1 (w3.org)](https://www.w3.org/TR/WCAG21/).

### Expected behaviour

The segment title should be rendered as the appropriate heading. 

### CodeSandbox or Storybook URL

https://codesandbox.io/s/menu-headers-0655gh?file=/src/App.js:605-694

### JIRA Ticket (Sage Only)

SBS-47524

### Suggested Solution

There should be an option to pass a `headingType` prop to `MenuSegmentTitle `which sets the HTML heading element which is rendered on the page. 

### Carbon Version

117.4.0

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by sianford** [comment]
FE-5877
